test/common/alerts_exist.go

```diff
@@ -0,0 +1,530 @@
+		Rules: []string{
+			"3scaleRedisCacheConnectionFailed",
+		},
+	},
+	{
+		File: "redhat-rhmi-operator-connectivity-rule-threescale-backend-redis-example-rhmi.yaml",
+		Rules: []string{
+			"3scaleRedisCacheConnectionFailed",
+		},
+	},
+	{
+		File: "redhat-rhmi-operator-connectivity-rule-threescale-postgres-example-rhmi.yaml",
+		Rules: []string{
+			"3scalePostgresConnectionFailed",
+		},
+	},
+	{
+		File: "redhat-rhmi-operator-availability-rule-threescale-redis-example-rhmi.yaml",
+		Rules: []string{
+			"3scaleRedisCacheUnavailable",
+		},
+	},
+	{
+		File: "redhat-rhmi-operator-availability-rule-threescale-backend-redis-example-rhmi.yaml",
+		Rules: []string{
+			"3scaleRedisCacheUnavailable",
+		},
+	},
+	{
+		File: "redhat-rhmi-operator-availability-rule-threescale-postgres-example-rhmi.yaml",
+		Rules: []string{
+			"3scalePostgresInstanceUnavailable",
+		},
+	},
+	{
+		File: "redhat-rhmi-operator-connectivity-rule-ups-postgres-example-rhmi.yaml",
+		Rules: []string{
+			"upsPostgresConnectionFailed",
+		},
+	},
+	{
+		File: "redhat-rhmi-operator-availability-rule-ups-postgres-example-rhmi.yaml",
+		Rules: []string{
+			"upsPostgresInstanceUnavailable",
+		},
+	},
+	{
+		File: "redhat-rhmi-operator-availability-rule-codeready-postgres-example-rhmi.yaml",
+		Rules: []string{
+			"codeready-workspacesPostgresInstanceUnavailable",
+		},
+	},
+	{
+		File: "redhat-rhmi-operator-connectivity-rule-codeready-postgres-example-rhmi.yaml",
+		Rules: []string{
+			"codeready-workspacesPostgresConnectionFailed",
+		},
+	},
+	{
+		File: "redhat-rhmi-operator-connectivity-rule-rhssouser-postgres-example-rhmi.yaml",
+		Rules: []string{
+			"user-ssoPostgresConnectionFailed",
+		},
+	},
+	{
+		File: "redhat-rhmi-operator-availability-rule-rhssouser-postgres-example-rhmi.yaml",
+		Rules: []string{
+			"user-ssoPostgresInstanceUnavailable",
+		},
+	},
+	{
+		File: "redhat-rhmi-operator-connectivity-rule-rhsso-postgres-example-rhmi.yaml",
+		Rules: []string{
+			"rhssoPostgresConnectionFailed",
+		},
+	},
+	{
+		File: "redhat-rhmi-operator-availability-rule-rhsso-postgres-example-rhmi.yaml",
+		Rules: []string{
+			"rhssoPostgresInstanceUnavailable",
+		},
+	},
+}
+
+func TestIntegreatlyAlertsExist(t *testing.T, ctx *TestingContext) {
+	// get the RHMI custom resource to check what storage type is being used
+	rhmi := &v1alpha1.RHMI{}
+	ns := fmt.Sprintf("%soperator", namespacePrefix)
+	err := ctx.Client.Get(goctx.TODO(), types.NamespacedName{Name: installationName, Namespace: ns}, rhmi)
+	if err != nil {
+		t.Fatal("error getting RHMI CR:", err)
+	}
+
+	// add external database alerts to list of expected rules if
+	// cluster storage is not being used
+	if rhmi.Spec.UseClusterStorage != "true" {
+		for _, rule := range expectedAWSRules {
+			expectedRules = append(expectedRules, rule)
+		}
+	}
+
+	// exec into the prometheus pod
+	output, err := execToPod("curl localhost:9090/api/v1/rules",
```

Comment:
Wouldn't it be better to use `port-forward` to create a local proxy to the pod and then hit it using the golang `http` instant of executing curl inside the pod?

Reply:
I can look into it, why would this be a better way of doing it rather than doing an exec?

Reply:
I would question the benefit in changing this implementation without any benefit. Using port-forward sounds more cumbersome and prone to error also

Reply:
I don't have strong opinion for this, but I think that if we could reach the api directly it would simplify the code, as we would not have to deal with streams, stdout and stderr and the `execToPod` method wouldn't be necessary anymore but there is no easy way to use `port-forwarding` too

Reply:
Would like to mark this ad resolved but somehow I can't, same for the other two comments